### ISSUE TITLE
Could integrate with neovim?

### ISSUE THREAD

**Issue by myromeu** [JBakamovic/yavide#39]

**Comment by JBakamovic** [comment]
If `neovim` is backwards compatible with `vim` then I suppose it could work.

**Comment by TamaHobbit** [comment]
How would we go about this? It's unclear to me from the [docs](https://github.com/JBakamovic/yavide/tree/master/docs) what yavide actually is - seems to be more than a collection of plugins, as you have to compile it when installing; and yet, it depends on your version of vim, rather than using its source code in the project. Could you elaborate, e.g. on [the introduction page](https://github.com/JBakamovic/yavide/blob/master/docs/intro.md), how yavide is set up, so that we can see how to extend it or port it to neovim?

**Comment by TamaHobbit** [comment]
By the way, changing the alias the installation puts in your ~/.bashrc file to use vim instead of gvim allows you to run yavide through the terminal:

```
alias yavide="vim --servername yavide -f -N -u /opt/yavide/.vimrc"
```

that seems to work fine but the client-server stuff is different in nvim, it outputs an error:

```
nvim: Unknown option argument: "--servername"
```

**Comment by JBakamovic** [comment]
> It's unclear to me from the docs what yavide actually is - seems to be more than a collection of plugins ...

That is correct. Besides the pre-selected third-party plugins it also implements its own API to support asynchronous processing. It implements improved (semantic) syntax highlighting, compiling, automatic code formatting and indexing. Implementation of more services is yet to come.

> it depends on your version of vim

It does not. I'm not sure where did you get this from? I think the only requirement would be that vim version has to be at least 7.3. 

> Could you elaborate, e.g. on the introduction page, how yavide is set up, so that we can see how to extend it or port it to neovim?

There is nothing magic about `yavide`. It is an ordinary vim configuration and someone familiar with `vimscript` will find nothing unusual there. Extra stuff which is not found in other vim configurations is framework for async processing and services built on top of that framework.

**Comment by JBakamovic** [comment]
> By the way, changing the alias the installation puts in your ~/.bashrc file to use vim instead of gvim allows you to run yavide through the terminal

`vim` is different from `gvim` in couple of command line parameters that are crucial for implementing some of the `yavide` functionality and one of them for example being the `servername`. It is not supported by `vim`.

> that seems to work fine but the client-server stuff is different in nvim, it outputs an error

See above.

**Comment by TamaHobbit** [comment]
Ok; so "depends on your version of vim" should be "it uses whatever gvim installation you have, which should be at least 7.3" ?

The point is, neovim is a fork of vim that supports asynchronous processing, but does work through a terminal. So I think that porting the functionality mentioned (e.g. automatic code formatting and indexing) to neovim would most likely give us a better version. There's also a good chance that [the stability issues I've seen](https://github.com/JBakamovic/yavide/issues/37) are related to gvim, not yavide, and would therefore also be solved with a neovim port.

**Comment by JBakamovic** [comment]
Well, async processing in `yavide` is implemented via `clientserver` architecture and as far as I can tell `neovim` does not support it. To port the functionality you would need to replace it with the alternative available in `neovim`. [Stability issues](https://github.com/JBakamovic/yavide/issues/37) that you personally have described are really really strange and if you try to give me more details maybe I can be of more help.
